### ISSUE TITLE
Insane Lag

### ISSUE THREAD

**Issue by tookender** [Astromations/Hazy#141]
How do I make the application use more FPS or something? It is extremely laggy when scrolling down or navigation the application at all.

https://github.com/user-attachments/assets/628306b9-72d1-4cfa-b4c4-edfedf5e3f83

**Comment by Astromations** [comment]
This video isn't loading for me, but I'd recommend toggling Hardware Acceleration.

**Comment by tookender** [comment]
> This video isn't loading for me, but I'd recommend toggling Hardware Acceleration.
> 
> ![image](https://private-user-images.githubusercontent.com/80211195/398690459-cc8d486f-8078-446e-ad8b-c004a39e3d63.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0MDEzMDAsIm5iZiI6MTczNTQwMTAwMCwicGF0aCI6Ii84MDIxMTE5NS8zOTg2OTA0NTktY2M4ZDQ4NmYtODA3OC00NDZlLWFkOGItYzAwNGEzOWUzZDYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI4VDE1NTAwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMmQzMDhlYzU1MTc1NDg3YWYwYTNkZDA3ZGU1ZTk1NGVlZTQ3YmQwYmI1YWNlYTkxNDViYzA5ZTM3MzhjNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ob6YH6uD3I_EkYWyXb3OEOA0ceRbAbHb3zrnEvy94Do)

That unfortunately does nothing.

**Comment by JMTIEM** [comment]
Currently this theme uses waaaaay too much CPU, I guess it's just an optimization issue. It's quite disapointing, since the theme is the best one I ever found, but the lag issues are sometimes too overwhelming to use.
Let's hope the dev improves it, because it really is a nice theme!

**Comment by SoCuul** [comment]
The simple answer is your computer isn't powerful enough to handle all the CSS transparency/blur effects in-use by the theme. The same exact thing happened with my previous computer, and made the theme unusable for me. When I upgraded to a more powerful system, I haven't had any performance issues since. The effects are cpu-rendered at the moment though.

Apparently, you can potentially force elements to be rendered by the GPU (which would greatly improve performance), but I have not tested this myself: https://graffino.com/til/how-to-fix-filter-blur-performance-issue-in-safari

**Comment by JMTIEM** [comment]
It's strange though, my CPU is not bad at all, but if I check task manager whenever spotify is open (not minimized) it takes 30% of my CPU usage.
I will try to force it run through my GPU and see tho, thanks!

**Comment by SoCuul** [comment]
My spotify averages at 4% cpu usage when idle, and while scrolling/loading stuff like the hope page it goes up to a maximum of 12%

**Comment by tookender** [comment]
> The simple answer is your computer isn't powerful enough to handle all the CSS transparency/blur effects in-use by the theme. The same exact thing happened with my previous computer, and made the theme unusable for me. When I upgraded to a more powerful system, I haven't had any performance issues since. The effects are cpu-rendered at the moment though.
> 
> Apparently, you can potentially force elements to be rendered by the GPU (which would greatly improve performance), but I have not tested this myself: https://graffino.com/til/how-to-fix-filter-blur-performance-issue-in-safari

I am not sure that is the issue, I have a pretty powerful computer with a RTX 3080 Ti and a 9 5950X, I am pretty sure that should be able to handle some transparency/blur effects on Spotify, as it handles them on any other app such as my browser, fire explorer, etc.

**Comment by SoCuul** [comment]
> > The simple answer is your computer isn't powerful enough to handle all the CSS transparency/blur effects in-use by the theme. The same exact thing happened with my previous computer, and made the theme unusable for me. When I upgraded to a more powerful system, I haven't had any performance issues since. The effects are cpu-rendered at the moment though.
> > Apparently, you can potentially force elements to be rendered by the GPU (which would greatly improve performance), but I have not tested this myself: https://graffino.com/til/how-to-fix-filter-blur-performance-issue-in-safari
> 
> I am not sure that is the issue, I have a pretty powerful computer with a RTX 3080 Ti and a 9 5950X, I am pretty sure that should be able to handle some transparency/blur effects on Spotify, as it handles them on any other app such as my browser, fire explorer, etc.

That's quite strange then. I'm personally running it on an Apple M4 Max 128G so there may be some platform differences.

**Comment by iHelops** [comment]
Hi, I too have encountered this problem, although my PC is far from being the weakest. It's definitely not a theme issue, as the lags happen without it. Apparently Spotify redraws the screen too often when scrolling, which is causing the lags. I came across [a way](https://blog.aziz.tn/2025/01/spotify-fix-lagging-issue-on-scrolling.html) to make the browser redraw elements using the GPU, which greatly improved scrolling performance. For convenience, I made an extension that can be found in the spicetify shop called [`Smooth Scrolling`](https://github.com/iHelops/smooth-scrolling). Try this method, I think it will work for you

**Comment by sanoojes** [comment]
https://discord.com/channels/842219447716151306/1207377288581218465/1331768826278449184
![Image](https://github.com/user-attachments/assets/0a194829-13ec-46de-96bb-eb0650693d95)

He created an extension with AI that adds `transform` and `translate3d` properties to scrollable containers, which helps to reduce lag. I also made a simple CSS snippet that achieves a similar effect.

```css
div[role="tablist"],
.StyledTabList-tablist-tabList,
div[data-overlayscrollbars-viewport],
div[data-overlayscrollbars],
.search-searchCategory-categoryGrid,
.main-yourLibraryX-libraryContainer,
.main-yourLibraryX-libraryFilter,
.main-navBar-mainNav,
.main-yourLibraryX-library {
	will-change: transform;
	transform: translate3d(0px, 0px, 0px);
}
```

**Comment by sanoojes** [comment]
> Hi, I too have encountered this problem, although my PC is far from being the weakest. It's definitely not a theme issue, as the lags happen without it. Apparently Spotify redraws the screen too often when scrolling, which is causing the lags. I came across [a way](https://blog.aziz.tn/2025/01/spotify-fix-lagging-issue-on-scrolling.html) to make the browser redraw elements using the GPU, which greatly improved scrolling performance. For convenience, I made an extension that can be found in the spicetify shop called [`Smooth Scrolling`](https://github.com/iHelops/smooth-scrolling). Try this method, I think it will work for you

ohh it works the same i think using the above css is better as it reduces complication

**Comment by iHelops** [comment]
> ohh it works the same i think using the above css is better as it reduces complication

Yep, that's a good solution too. The js solution ensures that the optimisation is applied to all scrolling containers, so you can be sure that the next spotify update won't break it, even though it's unlikely. But then again, your solution is also very good
